reporter/src/main/kotlin/model/EvaluatedModelBuilder.kt

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter.model
+
+import com.here.ort.model.CuratedPackage
+import com.here.ort.model.Identifier
+import com.here.ort.model.OrtIssue
+import com.here.ort.model.OrtResult
+import com.here.ort.model.Package
+import com.here.ort.model.PackageReference
+import com.here.ort.model.Project
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.RuleViolation
+import com.here.ort.model.ScanResult
+import com.here.ort.model.ScanSummary
+import com.here.ort.model.config.IssueResolution
+import com.here.ort.model.config.PathExclude
+import com.here.ort.model.config.RuleViolationResolution
+import com.here.ort.model.config.ScopeExclude
+import com.here.ort.model.utils.FindingsMatcher
+import com.here.ort.model.yamlMapper
+import com.here.ort.reporter.ReporterInput
+import com.here.ort.reporter.utils.StatisticsCalculator
+import com.here.ort.utils.ProcessedDeclaredLicense
+
+/**
+ * Builds an [EvaluatedModel] from [input].
+ */
+class EvaluatedModelBuilder(val input: ReporterInput) {
```

Comment:
For `testability` of the code using this class it'd be better to have a function (ReporterInput) -> EvaluatedModel rather than making this a constructor parameter. I'd propose to rename this class also to `EvaluatedModelMapper` or at least swap `builder` against `mapper`. With builder pattern I associate something like `AbcBuilder().addX(x1).addX(x2).setY(y1).build()` which does not really fit here imo.

Reply:
It used to be a builder when I started the implementation but I changed that later. I have renamed it to `EvaluatedModelMapper` but would prefer to keep the input as a class property to not have to pass it around because it is used in several functions.

Reply:
Consider solving that by creating a private inner class.